src/Listener.js

```diff
@@ -0,0 +1,63 @@
+const desktopIdle = require('desktop-idle');
+
+// We want the status of the user to be accurate.
+// To ensure this we need to frequently check the users status
+const DEFAULT_POLLING_TIMEOUT = 1000;
+
+class Listener {
+  constructor(id, timeToAway, callback) {
+    this.id = id;
+    this.timeToAway = timeToAway;
+    this.timeoutId = null;
+    this.callback = callback;
+    this.isAway = false;
+    this.shouldListen = true;
+  }
+
+  checkIsAway() {
+    let idleSeconds;
+
+    try {
+      idleSeconds = desktopIdle.getIdleTime();
+    } catch (error) {
+      this.callback(null, error);
+      this.scheduleCheckIsAway();
+      return;
+    }
+
+    const isAway = idleSeconds >= this.timeToAway;
+
+    if (!this.isAway && isAway) {
+      this.callback({
+        status: 'away',
+        seconds: idleSeconds,
+        id: this.id,
+      });
+
+      this.isAway = isAway;
+    } else if (this.isAway && !isAway) {
+      this.callback({
+        status: 'back',
+        seconds: idleSeconds,
+        id: this.id,
+      });
+
+      this.isAway = isAway;
+    }
+
+    this.scheduleCheckIsAway();
+  }
+
+  scheduleCheckIsAway() {
```

Comment:
I don't really think so, that function isn't actually listening for anything, its scheduling a check

Reply:
Hm, you could say the same about the class name `Listener`. This class is `isn't actually listening for anything`... 

`listen` is more domain specific to the naming of this class, and the way i see it, the `listen` method is setting up this listener (by setting up the poll).

Maybe there is a better name to represent what this object is doing?

Reply:
I suppose we could call the class StatusListener , which is a bit more descriptive, and then use the `statusListener.listen()` maybe?

Reply:
Had a think about this, I don't think `Listener` is the correct term to describe this object. It suggests it does something that is not doing (listening).

How about calling it `ActivityWatcher` / `StatusWatcher` / `PresenceWatcher`?

I think calling `xxxWatcher` describes the intent of this class much better (because it is watching the value returned from the `desktop-idle` and executing the callback based on that).

The API could then be as simple as:

```
ActivityWatcher.start()

ActivityWatcher.stop() 
```

Reply:
`StatusWatcher` would get my vote there 👍